sync/downstream.py

```diff
@@ -184,12 +184,22 @@ def next_try_push(self, try_cls=trypush.TrySyntaxCommit):
 
         latest_try_push = self.latest_valid_try_push
 
-        if latest_try_push and latest_try_push.status != "complete":
-            return
+        if latest_try_push:
+            if latest_try_push.status != "complete":
+                return
+            if len(self.try_pushes()) > 20 and latest_try_push.infra_fail:
```

Comment:
20 does seem like a lot, but if a branch has a lot of changes this could be hit with only one infra fail. We probably want some more nuanced check here.

Reply:
So my only idea is to count the number of consecutive try pushes that have been infra_fails. That should be conservative enough, yeah?